lib/fastlane/actions/testfairy.rb

```diff
@@ -0,0 +1,109 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      TESTFAIRY_BUILD_URL = :TESTFAIRY_BUILD_URL
+    end
+
+    # To share this integration with the other fastlane users:
+    # - Fork https://github.com/KrauseFx/fastlane
+    # - Clone the forked repository
+    # - Move this integration into lib/fastlane/actions
+    # - Commit, push and submit the pull request
+
+    class TestfairyAction < Action
+      def self.run(params)
+        require 'shenzhen'
+        require 'shenzhen/plugins/testfairy'
+
+        Helper.log.info 'Starting with ipa upload to TestFairy...'.green
+
+        client = Shenzhen::Plugins::TestFairy::Client.new(
+          params[:api_key]
+        )
+
+        return params[:ipa] if Helper.test?
+
+        response = client.upload_build(params[:ipa], params.values)
+        if parse_response(response)
+          Helper.log.info "Build URL: #{Actions.lane_context[SharedValues::TESTFAIRY_BUILD_URL]}".green
+          Helper.log.info "Build successfully uploaded to TestFairy.".green
+        else
+          raise 'Error when trying to upload ipa to TestFairy'.red
+        end
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.parse_response(response)
+        if response.body && response.body.key?('status') && response.body['status'] == 'ok'
+          build_url = response.body['build_url']
+
+          Actions.lane_context[SharedValues::TESTFAIRY_BUILD_URL] = build_url
+
+          return true
+        else
+          Helper.log.fatal "Error uploading to TestFairy: #{response.body}".red
+
+          return false
+        end
+      end
+      private_class_method :parse_response
+
+      def self.description
+        'Upload a new build to TestFairy'
+      end
+
+      def self.available_options
+        # Define all options your action supports.
+
+        # Below a few examples
+        [
+          FastlaneCore::ConfigItem.new(key: :api_key,
+                                       env_name: "FL_TESTFAIRY_API_KEY", # The name of the environment variable
+                                       description: "API Key for TestFairy", # a short description of this parameter
+                                       verify_block: proc do |value|
+                                         raise "No API key for TestFairy given, pass using `api_key: 'key'`".red unless (value and not value.empty?)
```

Comment:
Use `!` instead of `not`.<br>Don't use parentheses around the condition of an `unless`.